plugin.py

```diff
@@ -23,16 +39,38 @@ def name(cls) -> str:
         return cls.__name__.lower()
 
     @classmethod
-    def can_start(cls, _: sublime.Window, __: sublime.View,
-        folders: List[WorkspaceFolder], configuration: ClientConfig
+    def on_pre_start(
+        cls,
+        window: sublime.Window,
+        initiating_view: sublime.View,
+        workspace_folders: List[WorkspaceFolder],
+        resolved: ResolvedStartupConfig,
+    ) -> Optional[str]:
+        starting_dir, filename = os_path.split(initiating_view.file_name())
+        path = _find_flow_root_path(starting_dir)
+        binpath = _flow_bin_path(path)
+        resolved.command = [binpath, "lsp"]
+        workspace_folders = [
+            WorkspaceFolder.from_path(os_path.join(path, "")),
+        ]
+        return path
+
+    @classmethod
+    def can_start(
+        cls,
+        _: sublime.Window,
+        initiating_view: sublime.View,
+        folders: List[WorkspaceFolder],
+        configuration: ClientConfig,
```

Comment:
The search for the path should actually happen here IMO. So, find the flow path in this function.

If flow cannot be found, return an error string like "no flow binary found".

If there are no workspace folders (can happen), then also return something like "missing folders".

Then, set the workspace folders here:

```python
workspace_folders.clear()
workspace_folders.append(WorkspaceFolder.from_path(path))
```

In the `on_pre_start` callback, only set the command using the single workspace folder:

```python
resolved.command = [_flow_bin_path(workspace_folders[0].path), "lsp"]
```